src/iarc7_motion/iarc_tasks/go_to_roomba_task.py

```diff
@@ -0,0 +1,190 @@
+                    rospy.logerr(ex.message)
+                    return (TaskAborted(msg = 'Exception when looking up transform during go_to_roomba'),)
+
+                if(transStamped.transform.translation.z > self._MIN_MANEUVER_HEIGHT):
+                    """ quarantine"""
+                    # check if the roomba is in range
+                    if not self._check_roomba_in_sight():
+                        return (TaskAborted(msg='The provided roomba is not in sight of quad'),)
+                    elif self._on_ground():
+                        return (TaskDone(),)
+
+                    try:
+                        roomba_transform = self.topic_buffer.get_tf_buffer().lookup_transform(
+                                        'map',
+                                        self._roomba_id,
+                                        rospy.Time(0),
+                                        rospy.Duration(self._TRANSFORM_TIMEOUT))
+                        roomba_check_pos = self.topic_buffer.get_tf_buffer().lookup_transform(
+                                         'level_quad',
+                                         self._roomba_id,
+                                         rospy.Time(0),
+                                         rospy.Duration(self._TRANSFORM_TIMEOUT))
+                    except (tf2_ros.LookupException,
+                            tf2_ros.ConnectivityException,
+                            tf2_ros.ExtrapolationException) as ex:
+                        rospy.logerr('ObjectTrackTask: Exception when looking up transform')
+                        rospy.logerr(ex.message)
+                        return (TaskAborted(msg='Exception when looking up transform during go to roomba'),)
+                        
+
+
+                    # Create point at drones center
+                    stamped_point = PointStamped()
+                    stamped_point.point.x = 0
+                    stamped_point.point.y = 0
+                    stamped_point.point.z = 0
+
+                    # returns point distances of roomba to center point of the map
+                    self._roomba_point_map = tf2_geometry_msgs.do_transform_point(
+                                                            stamped_point, roomba_transform)
+
+                    self._roomba_point_level_quad = tf2_geometry_msgs.do_transform_point(
+                                                            stamped_point, roomba_check_pos)
+
+
+                    roomba_x_velocity = self._roomba_odometry.twist.twist.linear.x
+                    roomba_y_velocity = self._roomba_odometry.twist.twist.linear.y
+
+                    x_vel_target = self._roomba_point_map.point.x + roomba_x_velocity
+                    y_vel_target = self._roomba_point_map.point.y + roomba_y_velocity
+                    
+                    self._path_holder.reinitialize_translation_stop_planner(x_vel_target,
```

Comment:
Wouldn't this mean that the make it so that it could track a moving Roomba's position. I also played around with this quite a bit and this seemed to be the combination that worked

Reply:
Can you please clarify your remark?

Reply:
go to Roomba will not work if I just do the Roomba odometery (this meaning that I just do `self._roomba_odometry.twist.twist.linear.x`)

Reply:
Actually is there another property of self._roomba_odometry that will do what is above but require no transforms?